src/features/processPicker.ts

```diff
@@ -117,16 +129,19 @@ export class RemoteAttachPicker {
     }
 
     public static getRemoteOSAndProcesses(pipeCmd: string): Promise<AttachItem[]> {
-        
-        // If the client OS is NOT Windows, we need to escape '$(uname)' as otherwise it will be evaluated on the client side
-        // If the client OS is Windows, we don't want to do that as it will be sent as-is, and thus the if statement will be broken
-        const remoteUnameCommand = os.platform() !== "win32" ? "$\\(uname\\)" : "$(uname)";
 
         // Commands to get OS and processes
-        const command = `bash -c 'uname && if [ "${remoteUnameCommand}" == "Linux" ] ; then ${RemoteAttachPicker.linuxPsCommand} ; elif [ "${remoteUnameCommand}" == "Darwin" ] ; ` +
-            `then ${RemoteAttachPicker.osxPsCommand}; fi'`;
+        const command = `uname && if [ "$(uname)" == "Linux" ] ; then ${RemoteAttachPicker.linuxPsCommand} ; elif [ "$(uname)" == "Darwin" ] ; ` +
+            `then ${RemoteAttachPicker.osxPsCommand}; fi`;
+
+        // Create a temp file to redirect commands to the pipeProgram to solve quoting issues.
+        const tempFile = tmp.fileSync();
+        fs.write(tempFile.fd, command);
+
+        return execChildProcessAndOutputErrorToChannel(`${pipeCmd} < ${tempFile.name}`, null, RemoteAttachPicker._channel).then(output => {
+            // Remove temp file since the uname and ps commands have been executed.
+            tempFile.removeCallback();
```

Comment:
>tempFile [](start = 12, length = 8)

If we keep this, make sure to add a catch handler to clean it up in the case of failure as well.